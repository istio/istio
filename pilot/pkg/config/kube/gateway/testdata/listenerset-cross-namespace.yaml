apiVersion: gateway.networking.k8s.io/v1beta1
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: parent-gateway
  namespace: istio-system
spec:
  allowedListeners:
    namespaces:
      from: All
  addresses:
    - value: istio-ingressgateway
      type: Hostname
  gatewayClassName: istio
  listeners:
    - name: foo
      hostname: foo.com
      protocol: HTTP
      port: 80
---
apiVersion: gateway.networking.x-k8s.io/v1alpha1
kind: XListenerSet
metadata:
  name: single-entry-http
  namespace: ns1
spec:
  parentRef:
    name: parent-gateway
    namespace: istio-system
    kind: Gateway
    group: gateway.networking.k8s.io
  listeners:
    - name: first
      hostname: first.foo.com
      protocol: HTTP
      port: 80
---
apiVersion: gateway.networking.x-k8s.io/v1alpha1
kind: XListenerSet
metadata:
  name: same-ns-cert
  namespace: ns2
spec:
  parentRef:
    name: parent-gateway
    namespace: istio-system
    kind: Gateway
    group: gateway.networking.k8s.io
  listeners:
    - name: second
      hostname: second.foo.com
      protocol: HTTPS
      port: 443
      tls:
        mode: Terminate
        certificateRefs:
          - kind: Secret
            group: ""
            name: ns2-cert
---
apiVersion: gateway.networking.x-k8s.io/v1alpha1
kind: XListenerSet
metadata:
  name: cross-ns-cert
  namespace: ns2
spec:
  parentRef:
    name: parent-gateway
    namespace: istio-system
    kind: Gateway
    group: gateway.networking.k8s.io
  listeners:
    - name: allowed
      hostname: allowed.foo.com
      protocol: HTTPS
      port: 443
      tls:
        mode: Terminate
        certificateRefs:
          - kind: Secret
            group: ""
            name: ns3-cert
            namespace: ns3
    - name: denied
      hostname: denied.foo.com
      protocol: HTTPS
      port: 443
      tls:
        mode: Terminate
        certificateRefs:
          - kind: Secret
            group: ""
            name: ns4-cert
            namespace: ns4
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: ReferenceGrant
metadata:
  name: do-not-allow-cert-transitively
  namespace: ns4
spec:
  from:
    - group: gateway.networking.k8s.io
      kind: Gateway
      namespace: istio-system
  to:
    - group: ""
      kind: Secret
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: ReferenceGrant
metadata:
  name: allow-listenerset
  namespace: ns3
spec:
  from:
    - group: gateway.networking.x-k8s.io
      kind: XListenerSet
      namespace: ns2
  to:
    - group: ""
      kind: Secret