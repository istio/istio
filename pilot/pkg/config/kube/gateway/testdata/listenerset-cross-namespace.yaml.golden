apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/parent-gateway/foo.istio-system
  creationTimestamp: null
  name: parent-gateway-istio-autogenerated-k8s-gateway-foo
  namespace: istio-system
spec:
  servers:
  - hosts:
    - istio-system/foo.com
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent-namespace: istio-system
    internal.istio.io/parents: XListenerSet/single-entry-http/first.ns1
  creationTimestamp: null
  name: single-entry-http-istio-autogenerated-k8s-gateway-first
  namespace: ns1
spec:
  servers:
  - hosts:
    - ns1/first.foo.com
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent-namespace: istio-system
    internal.istio.io/parents: XListenerSet/cross-ns-cert/allowed.ns2
  creationTimestamp: null
  name: cross-ns-cert-istio-autogenerated-k8s-gateway-allowed
  namespace: ns2
spec:
  servers:
  - hosts:
    - ns2/allowed.foo.com
    port:
      name: default
      number: 443
      protocol: HTTPS
    tls:
      credentialName: kubernetes-gateway://ns3/ns3-cert
      mode: SIMPLE
---
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent-namespace: istio-system
    internal.istio.io/parents: XListenerSet/same-ns-cert/second.ns2
  creationTimestamp: null
  name: same-ns-cert-istio-autogenerated-k8s-gateway-second
  namespace: ns2
spec:
  servers:
  - hosts:
    - ns2/second.foo.com
    port:
      name: default
      number: 443
      protocol: HTTPS
    tls:
      credentialName: kubernetes-gateway://ns2/ns2-cert
      mode: SIMPLE
---
